### PR TITLE
tests, refactor: remove `NewRandomVMWithDataVolumeWithRegistryImport`

### PR DESCRIPTION
**Pull request by orenc1**
use `libvmi` and `libdv` directly to create VMs/VMIs whose DataVolumes are configured with registry import, and completely remove the `NewRandomVMWithDataVolumeWithRegistryImport()` function from `tests/utils.go`.

### What this PR does
Before this PR:

After this PR:

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

**Comment by orenc1** [comment]
/hold

need to move the VMI-with-DV creation to a util function at the storage package to avoid code duplication.

**Comment by akalenyu** [comment]
> /hold
> 
> need to move the VMI-with-DV creation to a util function at the storage package to avoid code duplication.

Check out https://github.com/kubevirt/kubevirt/pull/11476#issuecomment-2004446098

**Comment by orenc1** [comment]
introducing a new function `NewVMWithDataVolumeWithURL(imageUrl string, storageClass string)` in the storage package that simplifies the calling contexts where a VM with DV with import URL is required.
/unhold

File: tests/vm_test.go [hunk 1]
```diff
@@ -30,6 +30,8 @@ import (
 	"strings"
 	"time"
 
+	"kubevirt.io/kubevirt/tests/storage"
```
Comment (akalenyu) [inline tests/vm_test.go:33]: I don't think we want to go the cross-sig-import route

I have a PR open (although with some resistance) to expose a vmdv factory
https://github.com/kubevirt/kubevirt/pull/11670

**Pull request closed without merging**